tests/ExceptionBagTest.php

```diff
@@ -0,0 +1,131 @@
+use WtfPhp\JsonApiErrors\Tests\Fakes\TestResponse;
+
+class ExceptionBagTest extends TestCase
+{
+    protected ExceptionBag $bag;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->bag = new ExceptionBag();
+    }
+
+    /** @test */
+    public function itShouldAddOneException()
+    {
+        $this->assertTrue($this->bag->isEmpty());
+
+        $exception = new Exception('Something went wrong.');
+        $this->bag->add($exception);
+        $this->assertIsArray($this->bag->getAll());
+        $this->assertFalse($this->bag->isEmpty());
+        $this->assertCount(1, $this->bag->getAll());
+        $this->assertInstanceOf(Exception::class, $this->bag->getAll()[0]);
+    }
+
+    /** @test */
+    public function itShouldAddOneCustomError()
+    {
+        $this->assertTrue($this->bag->isEmpty());
+
+        $exception = new TestError();
+        $this->bag->add($exception);
+        $this->assertIsArray($this->bag->getAll());
+        $this->assertFalse($this->bag->isEmpty());
+        $this->assertCount(1, $this->bag->getAll());
+        $this->assertInstanceOf(TestError::class, $this->bag->getAll()[0]);
+    }
+
+    /** @test */
+    public function itShouldAddMultipleExceptions()
+    {
+        $this->assertTrue($this->bag->isEmpty());
+
+        $this->bag->addMultiple($this->getExceptions());
+        $this->assertIsArray($this->bag->getAll());
+        $this->assertFalse($this->bag->isEmpty());
+        $this->assertCount(3, $this->bag->getAll());
+        foreach ($this->bag->getAll() as $exception) {
+            $this->assertInstanceOf(Exception::class, $exception);
+        }
+    }
+
+    /** @test */
+    public function itShouldAddMultipleCustomErrors()
+    {
+        $this->assertTrue($this->bag->isEmpty());
+
+        $this->bag->addMultiple($this->getErrors());
+        $this->assertIsArray($this->bag->getAll());
+        $this->assertFalse($this->bag->isEmpty());
+        $this->assertCount(3, $this->bag->getAll());
+        foreach ($this->bag->getAll() as $error) {
+            $this->assertInstanceOf(TestError::class, $error);
+        }
+    }
+
+    /** @test */
+    public function itShouldClearBag()
+    {
+        $this->assertTrue($this->bag->isEmpty());
+
+        $this->bag->addMultiple($this->getExceptions());
+        $this->assertFalse($this->bag->isEmpty());
+        $this->bag->clear();
+        $this->assertTrue($this->bag->isEmpty());
+    }
+
+    /** @test */
+    public function itShouldNotAddNonThrowablesOnMultipleAdditions()
+    {
+        $this->assertTrue($this->bag->isEmpty());
+
+        $this->bag->addMultiple($this->throwablesAndNonthrowables());
+        $this->assertFalse($this->bag->isEmpty());
+        $this->assertCount(2, $this->bag->getAll());
+        foreach ($this->bag->getAll() as $exception) {
+            $this->assertInstanceOf(Exception::class, $exception);
+        }
+    }
+
+    /**
+     * @return array|Exception[]
```

Comment:
Why does the DocBlock say `array|Exception[]`? 
Either we always just specify the type with `array` or we also specify the type of the items in the array, but I wouldn't 

1. write down both in a Doc Block
2. do a mixture (as for e.g. `throwablesAndNonthrowables()` the return value is only declared as an array)